scripts/tool_benchmarks.py

```diff
@@ -0,0 +1,188 @@
+    "gpt-4-turbo-2024-04-09",
+    "accounts/fireworks/models/firefunction-v2",
+]
+
+examples = list(
+    client.list_examples(dataset_name="multiverse-math-examples-for-few-shot")
+)
+few_shot_messages = []
+questions = []
+for i, ex in enumerate(examples):
+    converted_messages = convert_to_messages(ex.outputs["output"])
+    questions.append(Document(converted_messages[1].content, metadata={"index": i}))
+    few_shot_messages += converted_messages
+
+few_shot_messages = [m for m in few_shot_messages if not isinstance(m, SystemMessage)]
+
+few_shot_str = ""
+for m in few_shot_messages:
+    if isinstance(m.content, list):
+        few_shot_str += "AI message: "
+        for tool_use in m.content:
+            if "name" in tool_use:
+                few_shot_str += f"Use tool {tool_use['name']}, input: {', '.join(f'{k}:{v}' for k,v in tool_use['input'].items())}"
+            else:
+                few_shot_str += tool_use["text"]
+            few_shot_str += "\n"
+    else:
+        if isinstance(m, HumanMessage):
+            few_shot_str += f"Human message: {m.content}"
+        else:
+            few_shot_str += f"AI message: {m.content}"
+
+    few_shot_str += "\n"
+
+vectorstore = Chroma.from_documents(
+    questions, embedding=OpenAIEmbeddings(model="text-embedding-3-small")
+)
+retriever = vectorstore.as_retriever()
+
+prompts = [
+    (
+        ChatPromptTemplate.from_messages(
+            [
+                ("system", "{instructions}"),
+                ("human", "{question}"),
+                ("placeholder", "{agent_scratchpad}"),
+            ]
+        ),
+        "zero-shot",
+    ),
+    (
+        ChatPromptTemplate.from_messages(
+            [
+                (
+                    "system",
+                    "{instructions} Here are some example conversations of the user interacting with the AI until the correct answer is reached: ",
+                ),
+                *few_shot_messages,
+                ("human", "{question}"),
+                ("placeholder", "{agent_scratchpad}"),
+            ]
+        ),
+        "few-shot-message",
+    ),
+    (
+        ChatPromptTemplate.from_messages(
+            [
+                (
+                    "system",
+                    "{instructions} Here are some example conversations of the user interacting with the AI until the correct answer is reached: "
+                    + few_shot_str,
+                ),
+                ("human", "{question}"),
+                ("placeholder", "{agent_scratchpad}"),
+            ]
+        ),
+        "few-shot-string",
+    ),
+    (
+        ChatPromptTemplate.from_messages(
+            [
+                (
+                    "system",
+                    "{instructions} Here are some example conversations of the user interacting with the AI until the correct answer is reached: ",
+                ),
+            ]
+            + semantic_similar_few_shots("{question}", retriever, examples)
```

Comment:
this needs to be part of the actual chain, currently its just getting static examples based on the string literal "{question}"

Reply:
should just have an examples placeholder thats filled at runtime